pkg/shipper/shipper_test.go

```diff
@@ -77,3 +79,85 @@ func TestShipperTimestamps(t *testing.T) {
 	testutil.Equals(t, int64(1000), mint)
 	testutil.Equals(t, int64(2000), maxt)
 }
+
+func TestIterBlockMetas(t *testing.T) {
+	var metas []*metadata.Meta
+	dir, err := ioutil.TempDir("", "shipper-test")
+	testutil.Ok(t, err)
+	defer func() {
+		testutil.Ok(t, os.RemoveAll(dir))
+	}()
+
+	for i := 0; i < 10; i++ {
+		id := ulid.MustNew(uint64(i), nil)
+		testutil.Ok(t, os.Mkdir(path.Join(dir, id.String()), os.ModePerm))
+		testutil.Ok(t,
+			metadata.Write(
+				log.NewNopLogger(),
+				path.Join(dir, id.String()),
+				&metadata.Meta{
+					BlockMeta: tsdb.BlockMeta{
+						ULID:    id,
+						MaxTime: int64((i + 1) * 1000),
+						MinTime: int64(i * 1000),
+						Version: 1,
+					},
+				},
+			),
+		)
+	}
+	rand.Shuffle(len(metas), func(i, j int) {
```

Comment:
I would vote for actually organizing initial metas ourselves. Relying on some random shuffling is not really deterministic.

Reply:
Ok.